tests/test_extension_image.py

```diff
@@ -8,40 +8,48 @@
 )
 
 
+actual_png = base64.b64decode(
+    b"iVBORw0KGgoAAAANSUhEUgAAADIAAAAyBAMAAADsEZWCAAAAG1BMVEXMzMy"
+    b"Wlpaqqqq3t7exsbGcnJy+vr6jo6PFxcUFpPI/AAAACXBIWXMAAA7EAAAOxA"
+    b"GVKw4bAAAAQUlEQVQ4jWNgGAWjgP6ASdncAEaiAhaGiACmFhCJLsMaIiDAE"
+    b"QEi0WXYEiMCOCJAJIY9KuYGTC0gknpuHwXDGwAA5fsIZw0iYWYAAAAASUVO"
+    b"RK5CYII="
+)
+actual_svg = (
+    '<?xml version="1.0" encoding="UTF-8"?>'
+    '<svg viewBox="0 0 50 50" xmlns="http://www.w3.org/2000/svg">'
+    '<g><rect width="50" height="50" fill="#fff"/>'
+    '<g><g fill="#fff" stroke="#707070">'
+    '<rect width="50" height="50" stroke="none"/>'
+    '<rect x="0" y="0" width="50" height="50" fill="none"/></g>'
+    '<text transform="translate(10 27)" fill="#707070" '
+    'font-family="ConsolasForPowerline, Consolas for Powerline" font-size="8">'
+    '<tspan x="0" y="0">50 x 50</tspan></text></g></g></svg>'
+)
+
+
 @pytest.fixture
 def snapshot_png(snapshot):
     return snapshot.use_extension(PNGImageSnapshotExtension)
 
 
-def test_image(snapshot_png, snapshot_svg):
-    actual_png = base64.b64decode(
-        b"iVBORw0KGgoAAAANSUhEUgAAADIAAAAyBAMAAADsEZWCAAAAG1BMVEXMzMy"
-        b"Wlpaqqqq3t7exsbGcnJy+vr6jo6PFxcUFpPI/AAAACXBIWXMAAA7EAAAOxA"
-        b"GVKw4bAAAAQUlEQVQ4jWNgGAWjgP6ASdncAEaiAhaGiACmFhCJLsMaIiDAE"
-        b"QEi0WXYEiMCOCJAJIY9KuYGTC0gknpuHwXDGwAA5fsIZw0iYWYAAAAASUVO"
-        b"RK5CYII="
-    )
+def test_image(snapshot_png):
     assert actual_png == snapshot_png
 
 
-@pytest.fixture
-def snapshot_svg(snapshot):
-    return snapshot.use_extension(SVGImageSnapshotExtension)
+def test_image_vector(snapshot):
+    """
+    Example of creating a previewable svg snapshot
+    """
+    assert snapshot(extension_class=SVGImageSnapshotExtension) == actual_svg
 
 
-def test_image_vector(snapshot_svg):
+def test_multiple_snapshot_extensions(snapshot):
     """
-    Example of creating a previewable svg snapshot
+    Example of switching extension classes on the fly.
+    These should be indexed in order of assertion.
     """
-    actual_svg = (
-        '<?xml version="1.0" encoding="UTF-8"?>'
-        '<svg viewBox="0 0 50 50" xmlns="http://www.w3.org/2000/svg">'
-        '<g><rect width="50" height="50" fill="#fff"/>'
-        '<g><g fill="#fff" stroke="#707070">'
-        '<rect width="50" height="50" stroke="none"/>'
-        '<rect x="0" y="0" width="50" height="50" fill="none"/></g>'
-        '<text transform="translate(10 27)" fill="#707070" '
-        'font-family="ConsolasForPowerline, Consolas for Powerline" font-size="8">'
-        '<tspan x="0" y="0">50 x 50</tspan></text></g></g></svg>'
-    )
-    assert snapshot_svg == actual_svg
+    assert actual_svg == snapshot(extension_class=SVGImageSnapshotExtension)
+    assert actual_svg == snapshot  # last extension class specified is used
```

Comment:
I feel like that behaviour can be confusing. Perhaps it should function similar to snapshot.use_extension, where it effectively clones the snapshot instance

Reply:
If it clones it then it'll lose the context of assertion index, leading to duplicates

Reply:
we can clone assertion index. I find it strange that calling `snapshot` at the beginning of a test case can affect snapshot assertions towards the end of the test case. Strange side-effects

Reply:
So your preferred behaviour would be having this line use the default snapshot extension class? It does make it more "pure" but then requires a lot of cloning or state values. Since the snapshot fixture is bound to the test case function the limit of the side effects would only be in specific test, I don't think that's too confusing and it would allow for doingt things like #161 without a lot of cloning.

Reply:
@noahnu also I don't think we could track with cloning assertion index if each individual assertion in the test case is a new instance